ls1intum/Artemis
# `Quiz exercises`: Improve quiz edit page and quiz exercises datatable

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


This is a follow-up PR to [#4936](https://github.com/ls1intum/Artemis/pull/4936) that introduces bug fixes and usability improvements in accordance with the new Quiz Modes that were introduced.

### Description


- The status dropdown in Quiz Edit page is removed since it is not needed anymore with the introduction of Quiz Lifecycles.
- If a Start Time was set, the "Start Quiz" button would be invisible in the Quiz Lifecycles column. However this meant that if the instructor wanted to start the quiz right now, they would have to go to Quiz edit page and manually set the "Start Time" to now to start the quiz. We make sure that also for this case, the button stays available for user convenience.
- The "Hidden" state for quiz exercises is replaced with "Invisible".
- Added help icons to explain the difference between release date and start date according to quiz modes. 
- Fixed issue where the batch is not fetched in edit page of Synchronized mode quizzes causing Start Time to be shown as empty.
- Fixed issue where pressing `Start Now` button for "Invisible" quizzes starting the quiz without making them "Visible". This prevented the students from accessing the quiz.
- Added release date and due date to quiz exercise component table.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Quiz Exercise

1. Log in to Artemis
2. Navigate to Course Administration
3. Go to Exercises page
4. Edit a Quiz Exercise
5. Make sure that the Status Dropdown is not available anymore at the footer. (Lefthand side where the save button is present)
6. Set release date to future.
7. Go back to the exercise page and see that the Quiz's status is "Invisible".
8. Edit quiz
9. Keep "Synchronized" mode and assign a Start time in the future. Save the page.
10. Refresh the page and make sure the "Start time" is populated correctly from the server.
11. Go back to the exercises page and click "Start Quiz".
12. Check status is "Active" and "Start Quiz" & "Set Visible" buttons are hidden.
13. Check release date, start date and due date columns are visible and have correct dates.
14. Make sure student can participate in the quiz.
15. Go to edit quiz again and even though the page is disabled, make sure that the "Start Time" is set to now.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage
Same as before





### Screenshots


![image](https://user-images.githubusercontent.com/67344158/167393521-955e63f9-3b78-4dec-a43a-ba5991daef12.png)
![image](https://user-images.githubusercontent.com/67344158/167393593-4006e057-610f-4865-a064-512ac1af37de.png)
![image](https://user-images.githubusercontent.com/67344158/167396429-f52580de-e38e-45c0-84f7-ad4e36f91989.png)
